x-pack/plugins/fleet/common/services/validate_package_policy.ts

```diff
@@ -234,6 +234,23 @@ export const validatePackagePolicyConfig = (
     }
   }
 
+  if (varDef.secret === true && parsedValue && parsedValue.isSecretRef === true) {
```

Comment:
Validation updated here to accept a secret reference in place of a value, e.g mongo password is a password field, but once a policy is created we only have a secret reference, so we cannot validate the field in the normal way.

Reply:
See the updated unit test above for an example